### PR TITLE
Implement find_next_gap

### PR DESCRIPTION
**Pull request by g2p**
This finds gaps in the sequence much faster than by just repeatedly calling mem.

### CODE REVIEW

**Comment by g2p** [comment]
By the way, we could easily add a similar find_next_interval function if the need arises.
The signature would be slightly different: it would return an interval option.

**Comment by g2p** [comment]
(I don't know if find_next_gap is the right name, find_closest_gap might be better)

File: lib/diet.ml [hunk 1]
```diff
@@ -561,6 +575,20 @@ module Test = struct
 
   let test_depth () = check_depth 1048576
 
+  let test_find_next () =
+    let open IntDiet in
+    let set = add (9, 9) @@ add (5, 7) empty in
+    assert (find_next_gap 0 set = 0);
```
Comment (avsm) [inline lib/diet.ml:581]: I think these `assert`s should be `OUnit2.assert_bool` (which will survive being compiled with `-noassert`)

**Review by avsm — CHANGES_REQUESTED** [review]

File: lib/diet.ml (hunk 1 above)
Comment (g2p) [reply lib/diet.ml:581]: Good point, but I went with the local style.
Should I switch the whole suite to OUnit2 asserts?

Comment (avsm) [reply lib/diet.ml:581]: Yes please, I think that'll be much better

Comment (samoht) [reply lib/diet.ml:581]: ounit?? nowadays you should use alcotest :p

Comment (samoht) [reply lib/diet.ml:581]: Although that would be a bit strange to use ounit/alcotest inside a library. Is there any way to test some of these invariants using the public API ?

Comment (g2p) [reply lib/diet.ml:581]: Changing the test style doesn't have much to do with this pull request.
Also, the invariants seem to rely on internal metadata.
Could this change be punted on for now?

Comment (avsm) [reply lib/diet.ml:581]: yes, that's fine. thanks for the PR!

**Review by avsm — APPROVED** [review]

**Pull request merged**